File: api/src/main/java/com/api/app/emum/PAY002.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 결제방식코드: payWayCode
 * */
@Getter
@AllArgsConstructor
public enum PAY002 implements CommonCode {
    CREDIT_CARD("001", "신용카드", 1, "", ""),
    POINT("002", "포인트", 2, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static PAY002 findByCode(String code) {
        return CommonCodeUtil.findByCode(PAY002.class, code);
    }
}

File: api/common-lib/common-core/src/main/kotlin/com/api/app/emum/PAY002.kt
package com.api.app.emum

/** 결제방식코드: payWayCode */
enum class PAY002(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    CREDIT_CARD("001", "신용카드", 1, "", ""),
    POINT("002", "포인트", 2, "", ""),
    ;

    companion object {
        fun findByCode(code: String): PAY002? = CommonCodeUtil.findByCode(PAY002::class.java, code)
    }
}
